components/Comments/ReportPost.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useRef, useState,useEffect } from 'react';
+import Flag from '../../icons/flag.svg'
+import { XIcon } from "@heroicons/react/outline";
+import toast from "react-hot-toast";
+import { signIn, useSession } from 'next-auth/react';
+import { createReportEmailTemplate } from '../../utils/createReportEmailTemplate';
+
+interface Props {
+    name:string;
+    body:string;
+    id:number;
+    email:string | null;
+    slug:string
+}
+
+export const ReportPost = (props:Props) => {
+
+  const { data:session } = useSession(); 
+  const { name, body, id, email, slug } = props;
+  const [isModalOpen, setModalOpen] = useState<boolean>(false);
+  const [comment, setComment] = useState('')
+
+  const handleOpenModal = () => setModalOpen(true);
+  const handleCloseModal = () => setModalOpen(false);   
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDialogElement>) => {
+    if (event.key === "Escape") {
+        handleCloseModal();
+    }
+  };
+
+  const handleSubmit = async (e:React.FormEvent) =>{
+    e.preventDefault();
+
+    const reportDetails = {
+        reportedById:session?.user?.id,
+        reportedByEmail:session?.user?.email,
+        reportedByUser:session?.user?.name,
+        reportedOnName:name,
+        reportedOnEmail:email,
+        reportedComment:body,
+        commentMadeByReporter:comment,
+        commentId:id,
+        timeReportSent:new Date(),
+        postLink:`https://codu.co/articles/${slug}`
+    }
+    
+    const htmlMessage = createReportEmailTemplate(reportDetails);
+
+    try {
+      await fetch('/api/sendEmail', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          recipient: process.env.ADMIN_EMAIL || '',
```

Comment:
I am wondering if we can move this into an api route and have the client side logic call the api route instead? this way client is not accessing env variables

Reply:
I knew this was not right. The problem is I dont have a good understanding of the pages router and trpc. Im going to take a few days to get up to speed on them and return to this with a better understanding.